### PR TITLE
Document custom load balancer usage

### PR DESCRIPTION
**Pull request by ndeloof**
### Proposed changes
Document how to create and use an external loadbalancer with ECS integration

### CODE REVIEW

File: engine/context/ecs-integration.md [hunk 1]
```diff
@@ -348,7 +348,42 @@ of a VPC when deploying a Compose application.
 - Use `x-aws-loadbalancer` as a top-level element in your Compose file to set
 the ARN of an existing LoadBalancer.
 
-- Use `external: true` inside a network definition in your Compose file for
+The latter can be used for those who want to customize application exposure, typically to
+use an existing domain name for your application:
+
+1. Use AWS web console or CLI to get your VPC and Subnets IDs. You can typically retrieve default VPC ID and attached subnets using this AWS CLI commands:
```
Comment (chris-crone) [inline engine/context/ecs-integration.md:354]:
```suggestion
1. Use the AWS web console or CLI to get your VPC and Subnets IDs. You can retrieve the default VPC ID and attached subnets using this AWS CLI commands:

```

File: engine/context/ecs-integration.md [hunk 2]
```diff
@@ -348,7 +348,42 @@ of a VPC when deploying a Compose application.
 - Use `x-aws-loadbalancer` as a top-level element in your Compose file to set
 the ARN of an existing LoadBalancer.
 
-- Use `external: true` inside a network definition in your Compose file for
+The latter can be used for those who want to customize application exposure, typically to
+use an existing domain name for your application:
+
+1. Use AWS web console or CLI to get your VPC and Subnets IDs. You can typically retrieve default VPC ID and attached subnets using this AWS CLI commands:
+```console
+➜  aws ec2 describe-vpcs --filters Name=isDefault,Values=true --query 'Vpcs[0].VpcId'
```
Comment (chris-crone) [inline engine/context/ecs-integration.md:356]:
```suggestion
$ aws ec2 describe-vpcs --filters Name=isDefault,Values=true --query 'Vpcs[0].VpcId' 

```

File: engine/context/ecs-integration.md [hunk 3]
```diff
@@ -348,7 +348,42 @@ of a VPC when deploying a Compose application.
...
 
-- Use `external: true` inside a network definition in your Compose file for
+The latter can be used for those who want to customize application exposure, typically to
+use an existing domain name for your application:
+
+1. Use AWS web console or CLI to get your VPC and Subnets IDs. You can typically retrieve default VPC ID and attached subnets using this AWS CLI commands:
+```console
+➜  aws ec2 describe-vpcs --filters Name=isDefault,Values=true --query 'Vpcs[0].VpcId' 
+"vpc-123456"
+➜  aws ec2 describe-subnets --filters Name=vpc-id,Values=vpc-123456 --query 'Subnets[*].SubnetId'
```
Comment (chris-crone) [inline engine/context/ecs-integration.md:358]:
```suggestion
$ aws ec2 describe-subnets --filters Name=vpc-id,Values=vpc-123456 --query 'Subnets[*].SubnetId'

```

File: engine/context/ecs-integration.md [hunk 4]
```diff
@@ -348,7 +348,42 @@ of a VPC when deploying a Compose application.
...
+[
+    "subnet-1234abcd", 
+    "subnet-6789ef00", 
+]
+```
+1. Use AWS CLI to create your load balancer. The AWS Web Console can also be used but will require to assing at least one listener, which we don't need here.
+```console
+➜  tutu aws elbv2 create-load-balancer --name myloadbalancer --type application --subnets "subnet-1234abcd" "subnet-6789ef00"
```
Comment (chris-crone) [inline engine/context/ecs-integration.md:366]:
```suggestion
$ aws elbv2 create-load-balancer --name myloadbalancer --type application --subnets "subnet-1234abcd" "subnet-6789ef00"

```

File: engine/context/ecs-integration.md [hunk 5]
```diff
@@ -348,7 +348,42 @@ of a VPC when deploying a Compose application.
...
+```console
+➜  aws ec2 describe-vpcs --filters Name=isDefault,Values=true --query 'Vpcs[0].VpcId' 
+"vpc-123456"
+➜  aws ec2 describe-subnets --filters Name=vpc-id,Values=vpc-123456 --query 'Subnets[*].SubnetId'
+[
+    "subnet-1234abcd", 
+    "subnet-6789ef00", 
+]
+```
+1. Use AWS CLI to create your load balancer. The AWS Web Console can also be used but will require to assing at least one listener, which we don't need here.
```
Comment (chris-crone) [inline engine/context/ecs-integration.md:364]:
```suggestion
1. Use the AWS CLI to create your load balancer. The AWS Web Console can also be used but will require adding at least one listener, which we don't need here.

```

File: engine/context/ecs-integration.md [hunk 6]
```diff
@@ -348,7 +348,42 @@ of a VPC when deploying a Compose application.
...
+1. To assign your application an existing domain name, you can configure your DNS with a
+CNAME entry pointing to just-created loadbalancer's `DNSName` reported as you created the loadbalancer.
+
+1. Use Loadbalancer ARN to set `x-aws-loadbalancer` in your compose file, and deploy your application using `docker compose up` command.
+
+Please note Docker ECS integration won't be aware of this domain name, so `docker compose ps` command will report URLs with loadbalancer DNSName, not your own domain.
+
+
+
+
```
Comment (chris-crone) [inline engine/context/ecs-integration.md:385]:
```suggestion
```

File: engine/context/ecs-integration.md [hunk 7]
```diff
@@ -360,6 +395,7 @@ networks:
     name: "sg-1234acbd"
 ```
 
+
```
Comment (chris-crone) [inline engine/context/ecs-integration.md:398]:
```suggestion
```

**Review by chris-crone — APPROVED** [review]
Minor changes then LGTM

**Review by gtardif — APPROVED** [review]

**Comment by usha-mandya** [comment]
Thank you @chris-crone.  Applied suggestions from the review

**Review by usha-mandya — APPROVED** [review]
LGTM

**Pull request merged**